### ISSUE TITLE
Changing the constraint solver

### ISSUE THREAD

**Issue by Sahandi78** [lsw9021/MASS#50]
Hi, I am working with this environment for training an exoskeleton attached to this human model. 

When I run the training script I get this messages:
`Error [BoxedLcpConstraintSolver.cpp:290] [BoxedLcpConstraintSolver] The solution of LCP includes NAN values: -nan -nan -nan -nan -nan -nan -nan -nan -nan -nan -nan -nan -nan -nan -nan -nan -nan -nan -nan -nan -nan -nan -nan -nan -nan -nan -nan. We're setting it zero for safety. Consider using more robust solver such as PGS as a secondary solver. If this happens even with PGS solver, please report this as a bug.
`

I want to know how can I change the solver to PGS? or is there any other solution?
I have seen this issue in dartsim repo : [https://github.com/dartsim/dart/issues/1058] but it did not solve the problem.

My exoskeleton is a URDF file which is loaded in the environment
Any help?

**Comment by JWAlmighty** [comment]
same problem here. the error occurs after 700 epochs
![image](https://github.com/user-attachments/assets/5850b4f9-e89e-48be-94bb-c65aaeaa2607)

**Comment by JWAlmighty** [comment]
turned out that the reward function in Environment.cpp is erroneous in the current commit e44c2ef. modified according to the paper
